Implement fit-to-track (#43)

mbof opened the pull request:
This is an attempt at implementing suggestion in issue #43.
I manually tested and it seemed to work ok.
Please feel free to disregard if this isn't done the right way.
Thanks!

vcoppe commented:
Cool! I have only looked at your code online but have a few comments:

- Most actions can be applied to a selection of files/tracks/segments/POIs, so here it would be nice to be able to make the map bounds fit any selection. Actually, the context menu can only be triggered from a selected item (if it is not, it becomes it, which simplifies the logic). The `$gpxStatistics` store contains the statistics of the current selection, so you can directly get the bounds of the current selection from there.
- Setting the same padding as I did elsewhere is nice. However, I think you can probably remove all other parameters to use the default animation?
- Isn't https://lucide.dev/icons/maximize nicer than https://lucide.dev/icons/maximize-2?
- I am not a native English speaker, but maybe "Fit on map" can be improved? Simply "Center map"?
- It could be nice to add a shortcut for this. Ideally, `Ctrl/Cmd+something` but it should not clash with the default shortcuts of major browsers. All shortcuts are captured in `Menu.svelte`, and it would probably be nicer to have a single function called from the button and shortcut, that you could put in `stores.ts` below `updateTargetMapBounds` (I should refactor this file at some point).

I hope this is clear, thank you very much for contributing!

mbof commented:
Thanks for the review! See updated PR with
* Use of `$gpxStatistics`
* Default animation (it's nicer, too — I didn't know there was one)
* Maximize icon instead of Maximize2 (no strong opinion here)
* Suggest "Fly to". "Center map" felt a bit weird to me, maybe "Center on map", but then the main menu version would be long and say "Center map on selection" which is pretty long vs. "Fly to selection". That said I don't feel strongly about it, feel free to change to whatever makes sense for you.
* Moved the logic to stores and added an entry in `Menu.svelte`. Eventually a keyboard shortcut can be added. I wasn't sure what would make sense because I don't really navigate with shortcuts myself, but I could see `Enter` being a reasonable choice? Not sure if you want to tackle this now or add the shortcut later.

Note - I tried to extend the `targetMapBounds` store to support this, but that seems to have barrier logic that only really makes sense during initialization and I found it a bit difficult to work around.
